Succulent/Classes/Succulent.swift

```diff
@@ -9,20 +9,36 @@
 import Embassy
 import Foundation
 
+public struct Configuration {
+    public var port: Int?
+    public var ignoreParameters: Set<String>?
+    public var ignoreVersioningRequests: [String]?
+    
+    public init() {
+        
+    }
+    
+    public init(port: Int? = nil, ignoreParameters: Set<String>? = nil, ignoreVersioningRequests: [String]?) {
+        self.port = port
+        self.ignoreParameters = ignoreParameters
+        self.ignoreVersioningRequests = ignoreVersioningRequests
+    }
+}
+
 public class Succulent : NSObject, URLSessionTaskDelegate {
     
-    public var port: Int?
-    public var version = 0
-    public var baseUrl: URL?
-    public var recordUrl: URL? {
+    public private(set) var port: Int?
+    public private(set) var version = 0
```

Comment:
Please remove the `private(set)` from `version`... I'm not sure that we actually need to expose this, but it isn't related to our configuration changes.